vault/request_handling.go

```diff
@@ -1922,17 +1920,24 @@ func (c *Core) handleDelegatedAuth(ctx context.Context, origReq *logical.Request
 			return authResp, nil, err
 		}
 	}
-	// A login request should never return a secret!
 	if authResp == nil {
 		return nil, nil, fmt.Errorf("%w: delegated auth request returned empty response for request_path: %s", ErrInternalError, authReq.Path)
 	}
+	// A login request should never return a secret!
 	if authResp.Secret != nil {
 		return nil, nil, fmt.Errorf("%w: unexpected Secret response for login path for request_path: %s", ErrInternalError, authReq.Path)
 	}
 	if authResp.Auth == nil || authResp.Auth.ClientToken == "" {
 		return nil, nil, fmt.Errorf("%w: delegated auth request did not return a client token for login path: %s", ErrInternalError, authReq.Path)
 	}
 
+	// Delegated auth tokens should only be batch tokens, as we don't want to incur
+	// the cost of storage/tidying for protocols that will be generating a token per
+	// request.
+	if !IsBatchToken(authResp.Auth.ClientToken) {
+		return nil, nil, fmt.Errorf("%w: delegated auth requests can only use batch tokens", logical.ErrPermissionDenied)
```

Comment:
maybe "must be configured to issue batch tokens"?

Reply:
Updated in https://github.com/hashicorp/vault/pull/23924/commits/1148b3d3d413ec4098d2b056a4330c9f2bc28f4c